pkg/core/chain/chain.go

```diff
@@ -355,9 +345,19 @@ func (c *Chain) AcceptBlock(ctx context.Context, blk block.Block) error {
 	}
 
 	// Caching the provisioners
-	// TODO: add bidList ?
+	// TODO: add bidList
 	c.p = &provisioners
 
+	// 3. Check the certificate
+	// This check should avoid a possible race condition between accepting two blocks
+	// at the same height, as the probability of the committee creating two valid certificates
+	// for the same round is negligible.
+	l.WithField("provisioners_len", c.p.Set.Len()).Trace("verifying block certificate")
+	if err := verifiers.CheckBlockCertificate(*c.p, blk); err != nil {
+		l.WithError(err).Error("certificate verification failed")
+		return err
+	}
+
```

Comment:
The order of execution seems wrong. If the certificate is invalid, we should not call `ExecuteStateTransition` function, since that function will mutate the smart contract storages with no possibility to rollback

Reply:
The wrong execution order is confirmed. In order to solve the race condition, a better solution has been found and is blocked by #672